src/Symfony/Bundle/FrameworkBundle/Logger/Logger.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Logger;
+
+use Psr\Log\AbstractLogger;
+use Psr\Log\InvalidArgumentException;
+use Psr\Log\LogLevel;
+
+/**
+ * Minimalist PSR-3 logger designed to write in stderr or any other stream.
+ *
+ * @internal
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+final class Logger extends AbstractLogger
+{
+    private static $levels = array(
+        LogLevel::DEBUG => 0,
+        LogLevel::INFO => 1,
+        LogLevel::NOTICE => 2,
+        LogLevel::WARNING => 3,
+        LogLevel::ERROR => 4,
+        LogLevel::CRITICAL => 5,
+        LogLevel::ALERT => 6,
+        LogLevel::EMERGENCY => 7,
+    );
+
+    private $minLevelIndex;
+    private $formatter;
+    private $handle;
+
+    public function __construct($minLevel = LogLevel::WARNING, $output = 'php://stderr', callable $formatter = null)
+    {
+        if (!isset(self::$levels[$minLevel])) {
+            throw new InvalidArgumentException(sprintf('The log level "%s" does not exist.', $minLevel));
+        }
+
+        $this->minLevelIndex = self::$levels[$minLevel];
+        $this->formatter = $formatter;
+        if (false === $this->handle = @fopen($output, 'a')) {
+            throw new InvalidArgumentException(sprintf('Unable to open "%s".', $output));
+        }
+    }
+
+    public function __destruct()
+    {
+        fclose($this->handle);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function log($level, $message, array $context = array())
+    {
+        if (!isset(self::$levels[$level])) {
+            throw new InvalidArgumentException(sprintf('The log level "%s" does not exist.', $level));
+        }
+
+        if (self::$levels[$level] < $this->minLevelIndex) {
+            return;
+        }
+
+        $formatter = $this->formatter;
+        $message = $formatter ? $formatter($level, $message, $context) : sprintf('[%s] %s', $level, $this->interpolate($message, $context)).\PHP_EOL;
+
+        fwrite($this->handle, $message);
+    }
+
+    /**
+     * Interpolates context values into the message placeholders.
+     *
+     * @author Jordi Boggiano <j.boggiano@seld.be>
+     *
+     * @see https://github.com/Seldaek/monolog/blob/master/src/Monolog/Processor/PsrLogMessageProcessor.php
+     *
+     * @param string $message
+     *
+     * @return string
+     */
+    private function interpolate($message, array $context)
+    {
+        if (false === strpos($message, '{')) {
+            return $message;
+        }
+
+        $replacements = array();
+        foreach ($context as $key => $val) {
+            if (null === $val || is_scalar($val) || (\is_object($val) && method_exists($val, '__toString'))) {
```

Comment:
Accessing function via the global namespace is inconsistent `\is_object` vs `is_scalar` etc. Any reasons for this?

Reply:
Yes: https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/3048 (`is_object` is in the list, `is_scalar` is not).